packages/reactive_ble_mobile/android/src/main/kotlin/com/signify/hue/flutterreactiveble/PluginController.kt

```diff
@@ -132,84 +157,113 @@ class PluginController {
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe(
+                { charResult ->
+                    when (charResult) {
+                        is com.signify.hue.flutterreactiveble.ble.CharOperationSuccessful -> {
+                            val charInfo = protoConverter.convertCharacteristicInfo(
+                                readCharMessage.characteristic,
+                                charResult.value.toByteArray()
+                            )
+                            charNotificationHandler.addSingleReadToStream(charInfo)
+                        }
+
+                        is com.signify.hue.flutterreactiveble.ble.CharOperationFailed -> {
                             protoConverter.convertCharacteristicError(
-                                    readCharMessage.characteristic,
-                                    throwable.message)
+                                readCharMessage.characteristic,
+                                "Failed to connect"
+                            )
                             charNotificationHandler.addSingleErrorToStream(
-                                    readCharMessage.characteristic,
-                                    throwable?.message ?: "Failure")
+                                readCharMessage.characteristic,
+                                charResult.errorMessage
+                            )
                         }
-                )
-                .discard()
+                    }
+                },
+                { throwable ->
+                    protoConverter.convertCharacteristicError(
+                        readCharMessage.characteristic,
+                        throwable.message
+                    )
+                    charNotificationHandler.addSingleErrorToStream(
+                        readCharMessage.characteristic,
+                        throwable?.message ?: "Failure"
+                    )
+                }
+            )
+            .discard()
     }
 
     private fun writeCharacteristicWithResponse(call: MethodCall, result: Result) {
-        executeWriteAndPropagateResultToChannel(call, result, com.signify.hue.flutterreactiveble.ble.BleClient::writeCharacteristicWithResponse)
+        executeWriteAndPropagateResultToChannel(
+            call,
+            result,
+            com.signify.hue.flutterreactiveble.ble.BleClient::writeCharacteristicWithResponse
+        )
     }
 
     private fun writeCharacteristicWithoutResponse(call: MethodCall, result: Result) {
-        executeWriteAndPropagateResultToChannel(call, result, com.signify.hue.flutterreactiveble.ble.BleClient::writeCharacteristicWithoutResponse)
+        executeWriteAndPropagateResultToChannel(
+            call,
+            result,
+            com.signify.hue.flutterreactiveble.ble.BleClient::writeCharacteristicWithoutResponse
+        )
     }
 
     private fun executeWriteAndPropagateResultToChannel(
-            call: MethodCall,
```

Comment:
this big change set makes me think we should enforce standard formatting rules so everyone formats the same

Reply:
Let's agree on a set (I think dartfmt, ktlint, swiftlint), th en add as part of CI.
Happy to have a stab. :-)